### DISCUSSION TITLE
Help with SumatraPDF iFilters

### DISCUSSION THREAD

**Discussion by mmontelatici** [sumatrapdfreader/sumatrapdf#2460] [Q&A]
Hi all,
I'm trying to use sumatraPdf iFilters in SqlServer Full Text Search (but also in windows indexing service) to search into PDF files, and it does not work.

I see that the filter is properly installed in windows indexing service:

![image](https://user-images.githubusercontent.com/38101293/161936749-69a818b4-097f-41e8-9acd-0bcc43ae08c9.png)

And also explicitly declared in SqlServer FTS configuration for PDF extensions (although version and manufacturer are missing):

![image](https://user-images.githubusercontent.com/38101293/161935505-c0f0b716-aa01-4ed7-a69e-8023b0ea8a1f.png)

But so far I was not able to find anything in PDF files.
Am I missing something in configuration?

I'm on a Windows 10 21H2 os, using SqlServer 2017. Thanks anybody who can help me.

**Comment by GitHubRulesOK** [comment]
I don't know how SQL may use that feature, 

and in a quick dubious folder files test with a fresh install in windows 11 the iFilter is very very slow as you need to set the iSearch to full contents and PDFs are a resource hog as they need fully expanding to memory from file system to do the internal search.

Its faster to use text extractors for content (many many pages per second) then the text stored in the database can be found with the pointer to the folder with the compressed pdf file (you don't keep the pdf archives files inside the sql archive do you?)

**Reply by mmontelatici** [reply]
> you don't keep the pdf archives files inside the sql archive do you?

The PDF documents, as well as other documents of any other type, are stored into a table with a FULLTEXT index accordingly to FTS specifications. 
All other doc types are correctly indexed and FTS works, but for SumatraPDF iFilter. When it is the iFilter library referenced by FTS, new PDF documents are not indexed and no results are found when searching.

**Comment by kjk** [comment]
I've never used or tested the IFilter in that scenario, so you're on your own debugging this.

**Comment by pthall** [comment]
On a new Windows 10 install, the Sumatra iFilter did not work for Windows indexing.

I ended up installing the old but working 64-bit Adobe iFilter:
http://ftp.adobe.com/pub/adobe/acrobat/win/11.x/PDFFilter64Setup.msi

More info:
https://web.archive.org/web/20201112015452/supportdownloads.adobe.com/detail.jsp?ftpID=5542

**Reply by mmontelatici** [reply]
Thank you for your answer, 
unfortunately Adobe iFilter has reached the End-Of-Support and I'm looking for a library that is still _alive_.

**Reply by GitHubRulesOK** [reply]
SumatraPDF iFilter should be working as I said I tested the current on latest windows 11 even using sandbox , 
however it works as designed within Windows Explorer search thus rules of engagement are 
- the correct bit ness must match e.g. on a 64bit system you must have installed and thus registered the 64bit dll (@mmontelatici you seem to have done that)
- it can only search inside a pdf if explorer Isearching is on for compressed contents in the folder with the files (same as if you were looking for text in a zip)
- It will be slow, that is not a bug, it is that unless the index is stored gigantically by windows, then ALL the PDF files need time for unpacking to find one word. so for testing ensure the text can be read by copy and paste (often words are broken inside the pdf as single letter or corrupted characters)

here I list across many folder there  are 70 files for searching
![image](https://user-images.githubusercontent.com/12754913/162181097-02bdd1a6-385f-4d96-a4be-b5b97f72f21d.png)

***I ensure the index will search inside the content (there is no point in search otherwise but not all pdf have searchable content)***

![image](https://user-images.githubusercontent.com/12754913/162181714-66fa328f-c65e-427f-8880-6f90d5793508.png)

***I then ask windows explorer to show me how many files contain Availability as a searchable word and the answer including duplicates is 7***

![image](https://user-images.githubusercontent.com/12754913/162182657-3a5b914c-7bd9-4d97-b547-c8d6d434af45.png)
